autokey-module/main.tf

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  default_suffix     = var.suffix != "" ? var.suffix : random_string.suffix.result
+  autokey_project_id = "${var.autokey_project_id}-${local.default_suffix}"
+  autokey_folder     = "${var.autokey_folder}-${local.default_suffix}"
+}
+
+resource "random_string" "suffix" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+resource "google_folder" "autokms_folder" {
+  count = var.create_autokey_folder ? 1 : 0
+
+  provider     = google-beta
+  display_name = local.autokey_folder
+  parent       = var.autokey_parent
+}
+
+module "autokey-project" {
+  count = var.create_autokey_project ? 1 : 0
+
+  source  = "terraform-google-modules/project-factory/google"
+  version = "~> 16.0"
+
+  project_id = var.create_autokey_project ? local.autokey_project_id : var.autokey_project_id
+  folder_id  = var.create_autokey_folder ? google_folder.autokms_folder[0].folder_id : var.autokey_folder
+
+  name            = var.autokey_project_name
+  billing_account = var.billing_account
+
+  activate_api_identities = [{
+    api = "cloudkms.googleapis.com"
+    roles = [
+      "roles/cloudkms.admin"
+    ]
+  }]
+
+  depends_on = [google_folder.autokms_folder]
+}
+
+data "google_project" "autokey_project" {
+  count = var.create_autokey_project ? 0 : 1
+
+  project_id = var.autokey_project_id
+}
+
+resource "google_project_service" "enable_kms_api" {
```

Comment:
In others blueprints in this repo, even when we have pre requirements regarding enabling APIs, we have this TF resource to ensure no errors for customers. (And we usually mention this behavior at the READMEs)

I'm OK to remove it if you think it's better, but this is not why we have `billing_account` being received in this module. We need billing because, in order to configure autokey, a KMS project can be created by `project-factory` when `create_autokey_project` is true.

So, in summary, in order to remove `billing_account` from the inputs of this module, we would need to cut the "auto creation project" feature and make required an existing KMS project to apply the module. (and also a resource project that will be introduced in the next PR)

Did the explanation make sense, WDYT?